src/pages/index.js

```diff
@@ -73,57 +67,196 @@ function handleImageClick(cardData) {
 const cardSection = new Section(
   {
     items: initialCards,
-    renderer: renderCard,
+    renderer: (item) => {
+      const cardELement = createCard(item);
+      cardSection.addItem(cardELement);
+    },
   },
-  cardListEl
+  ".cards__list"
 );
 
-cardSection.renderItems();
+//Event Listeners
 
-// Render Cards
-function renderCard(item, method = "addItem") {
-  const cardElement = createCard(item);
-  cardSection[method](cardElement);
-}
+addNewCardButton.addEventListener("click", () => {
+  addCardFormPopup.open();
+});
+
+profileEditButton.addEventListener("click", () => {
+  editProfileFormValidator.resetValidation();
+  const inputData = userInfo.getUserInfo();
+  // console.log(inputData);
+  profileNameInput.value = inputData.name;
+  profileDescriptionInput.value = inputData.about;
+  profileEditPopup.open();
+});
+
+addCardFormValidator.enableValidation();
+editProfileFormValidator.enableValidation();
 
 //Functions
 function createCard(cardData) {
-  const card = new Card(cardData, "#card-template", handleImageClick);
-  return card.getView();
+  const card = new Card(
+    cardData,
+    "#card-template",
+    handleImageClick,
+    handleDeleteCardSubmit,
+    likeCard,
+    unlikeCard
+  );
+  return card.generateCard();
 }
 
 function handleProfileEditSubmit(inputData) {
-  userInfo.setUserInfo({
-    name: inputData.name,
-    job: inputData.description,
-  });
+  profileEditPopup.renderLoading(true);
+
+  api
+    .setUserInfo(inputData.name, inputData.description)
+    .then(() => {
+      userInfo.setUserInfo({
+        name: inputData.name,
+        about: inputData.description,
+      });
+      profileEditPopup.close();
+    })
+    .catch((err) => {
+      console.error("Error updating user info", err);
+      alert(err);
+    })
+    .finally(() => {
+      console.log("Edit form complete");
+      profileEditPopup.renderLoading(false);
+    });
 
-  profileEditPopup.close();
+  // profileEditPopup.close();
 }
 
 function handleAddCardSubmit(inputData) {
-  const cardData = {
-    name: inputData.title,
-    link: inputData.url,
-  };
-  renderCard(cardData);
-  addCardFormValidator.disableButton();
-  addCardFormPopup.close();
-  addCardForm.reset();
+  const name = inputData.title;
+  const link = inputData.url;
+
+  addCardFormPopup.renderLoading(true);
+  api
+    .uploadCard({ name, link })
+    .then((cardData) => {
+      const card = createCard(cardData);
+
+      cardSection.addItem(card);
+      addCardFormPopup.close();
+      addCardForm.reset();
+      addCardFormValidator.resetValidation();
```

Comment:
`resetValidation` is for clearing possible validation errors when you open the **profile** form, because you insert **valid** data into its inputs. But when you add a new card, you don't have any errors after submitting. You need to call `toggleButtonState` there.